docs/jenkins.md

```diff
@@ -0,0 +1,190 @@
+# Jenkins
+
+The Foreman project maintains its own Jenkins instance for continuous integration at https://ci.theforeman.org/
+
+It runs the following types of tests:
+
+* Unit tests on the main develop and all supported stable branches of core Foreman projects
+* Pull request tests against core Foreman projects
+* Unit and pull request tests for many Foreman plugins
+* Nightly and release package building and release processes
+* Package smoke and integration testing
+* Package jobs for Pulpcore
+
+## Quick reference for maintainers
+
+Current PR test jobs (used on Foreman itself) support these commands:
+
+* `ok to test` - run tests for an unknown user, if the code within the patch is not malicious
+* `[test STATUS-NAME]`, e.g. `[test foreman]` to re-run a particular set of tests
+
+## Quick reference for plugin maintainers
+
+### Foreman plugin testing
+
+Foreman plugins are tested by adding the plugin to a Foreman checkout and running core tests, so it checks that existing behaviours still work and new plugin tests are run too.  The [test_plugin_matrix job](https://ci.theforeman.org/job/test_plugin_matrix/) copies the core jobs, but adds a plugin from a given git repo/branch and is usually used to test plugins in a generic way.
+
+Each plugin should have a job defined in JJB that calls test_plugin_matrix here: https://ci.theforeman.org/view/Plugins/
+
+#### Foreman plugin PR testing
+
+To test pull requests, a separate job is used that also takes the PR details: https://ci.theforeman.org/view/Plugins/job/test_plugin_pull_request/
+
+#### Adding a new Foreman plugin
+
+For a plugin "foreman_example", first create a job that tests the main (master or develop) branch.
+
+* ensure plugin tests (if any) run when `rake jenkins:unit` is called, see [the example plugin](https://github.com/theforeman/foreman_plugin_template/) and [testing a plugin](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Testing) for help
+* create a foreman_example.yaml file in [foreman-infra/JJB](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins)
+  * This will create a "test_plugin_foreman_example_master" job in Jenkins to test the master branch.
+* ensure the job is green by fixing bugs, installing dependencies etc.
+* add hook to GitHub repo, see [GitHub repo hook](https://projects.theforeman.org/projects/foreman/wiki/Jenkins#Testing-develop)
```

Comment:
This link still refers to the wiki, but this document is supposed to replace it.

Perhaps this should also at the org admin part since adding the Bots team is also a Github repo action.